app/controllers/admin/web_hooks_controller.rb

```diff
@@ -135,6 +136,26 @@ def redeliver_event
     end
   end
 
+  def redeliver_failed_events
+    web_hook_events =
+      @web_hook
+        .web_hook_events
+        .includes(:redelivering_webhook_event)
+        .not_ping
+        .where(id: params[:event_ids])
+
+    if web_hook_events
```

Comment:
I think instead of an `if` statement you should do a [guard clause](https://patrickkarsh.medium.com/ruby-refinements-unlocking-the-power-of-guard-clauses-for-streamlined-programming-5b60b2f2e6e), and return early if there are no events.

So something like
```ruby
raise Discourse::InvalidParameters if web_hook_events.count.zero?
```